schema/validator.go

```diff
@@ -86,13 +106,28 @@ func (v Validator) Validate(src io.Reader) error {
 	}
 }
 
-type unimplemented string
+// ValidateRefMedia validates the referenced objects against OCI media type defined in spec.
+// r: the given reader.
+// v: the expected media type.
+func ValidateRefMedia(r io.Reader, v Validator) error {
+	f, ok := vlidateRefMediaMap[v]
+	if ok {
+		if f == nil {
+			return fmt.Errorf("referenced media type validation %q unimplemented", v)
+		}
 
-func (v unimplemented) Validate(src io.Reader) error {
-	return fmt.Errorf("%s: unimplemented", v)
+		buf, err := ioutil.ReadAll(r)
+		if err != nil {
+			return errors.Wrap(err, "unable to read the document file")
+		}
+
+		return f(bytes.NewReader(buf))
+	}
+
+	return nil
```

Comment:
I think we should be raising the same unimplemented error here as you currently have for `f == nil`.  How about:

```
f, ok := vlidateRefMediaMap[v]
if ok && f != nil {
    return f(r)
}
return fmt.Errorf("referenced media type validation %q unimplemented", v)
```